### ISSUE TITLE
DBC Feeder uses TLS for connection to kuksa.val Server by default

### ISSUE THREAD

**Issue by sophokles73** [eclipse/kuksa.val.feeders#133]
The [README](https://github.com/eclipse/kuksa.val.feeders/blob/3a72e0b7230927205507e5fd96c5162cc26af56a/dbc2val/README.md?plain=1#L251) of the dbc2val feeder suggests that the feeder will not use TLS for connecting to kuksa.val Server/Databroker.

However, [ServerClientWrapper](https://github.com/eclipse/kuksa.val.feeders/blob/3a72e0b7230927205507e5fd96c5162cc26af56a/dbc2val/dbcfeederlib/serverclientwrapper.py#L30) actually seems to default to using TLS.

We should either adapt the documentation or the implementation accordingly.

**Comment by SebastianSchildt** [comment]
The default configuration is False: https://github.com/eclipse/kuksa.val.feeders/blob/3a72e0b7230927205507e5fd96c5162cc26af56a/dbc2val/config/dbc_feeder.ini#L33

**Comment by sophokles73** [comment]
Not if I provide my own config file that doesn't contain the *tls* option, or am I mistaken?

**Comment by SebastianSchildt** [comment]
Ah that _may_ be true, not sure without checking code. @erikbosch might know the answer immediately. He should be back really soon.

I agree, with the original issue: It should of course be consistent  between code and docs

**Comment by sophokles73** [comment]
Let me check this for you ;-)

https://github.com/eclipse/kuksa.val.feeders/blob/3a72e0b7230927205507e5fd96c5162cc26af56a/dbc2val/dbcfeeder.py#L483

**Comment by SebastianSchildt** [comment]
In my opinion it is good that "the default is TLS __unless__ you set it to false in config or cmd line", wile the default _configfile_ could still be "false", as realistically that is what most people will use.

It should just be reflected in the docs.

**Comment by sophokles73** [comment]
Fair enough, then I guess we should change https://github.com/eclipse/kuksa.val.feeders/blob/3a72e0b7230927205507e5fd96c5162cc26af56a/dbc2val/dbcfeederlib/databrokerclientwrapper.py#L49
and the README accordingly ...

**Comment by sophokles73** [comment]
I have taken another look at the supported command line switches and their defaults.
It seems strange to me that not all of the config properties can be configured using command line arguments and some of the implementation relies on default values set in a default config file included in the code tree instead of assuming reasonable defaults in the implementation itself.

For example, it seems to be impossible to disable usage of TLS on the command line, neither by means of a command line switch nor by means of an environment variable. Is that intended and follows a particular pattern (which I am not able to see) or is this simply a case of *not implemented yet*?

**Comment by erikbosch** [comment]
We have an issue at https://github.com/eclipse/kuksa.val/issues/589 on the fact that we are not that consistent in config options and the fact that we maybe should not support all config settings as command line arguments. There has also now and then been concerns that we should not change existing (default) behavior as that may cause regressions for downstream projects, I guess that is one of the reasons why broker and server had different default behavior. I will look more in detail and create a PR to better align behavior when time permits

**Comment by erikbosch** [comment]
We have migrated CAN Provider (dbcfeeder) to a new [repo](https://github.com/eclipse-kuksa/kuksa-can-provider). Long term my intention is to close all dbc/can-related issues in this repository. If we see the issue as still interesting or relevant we could create a new issue in the new repo, possibly referencing the old issue in the repository.

Suggested way forward:

- I will migrate selected issues early 2024
- If this issue has not been migrated but you want it to remain open please comment.
- Remaining issues no one seems to be interested in will be closed around 2024-03-31
